provider/haproxy/haproxy.go

```diff
@@ -310,17 +333,20 @@ func (dm *drainMgr) olderPidExists() (bool, error) {
 func (dm *drainMgr) AddEndpointForDrain(ep *config.Endpoint, pid string) bool {
 	//check if drain is needed by looking at Stats
 	if ep != nil && ep.Name != "" {
-		scur, pxname, err := dm.readCurrentStats(ep)
+		scur, pxnames, err := dm.readCurrentStats(ep)
 		if err != nil {
 			logrus.Errorf("AddEndpointForDrain: Error %v", err)
 			return false
 		}
 		//set weight to zero on the socket anyway to stop accepting new connections
```

Comment:
Just curious - why "anyway"?

Reply:
meant to say irrespective of whether Drain is needed on this backend or not - at this point we dont know that yet. Also weight will be set to "0" via config once metadata signals a change. But to avoid new connections in this gap (of metadata reload happening), we first set the weight to zero and then check if we should drain